wayshot/src/utils.rs

```diff
@@ -128,14 +133,45 @@ impl FromStr for EncodingFormat {
             "png" => Self::Png,
             "ppm" => Self::Ppm,
             "qoi" => Self::Qoi,
+            "webp" => Self::Webp,
             _ => bail!("unsupported extension '{s}'"),
         })
     }
 }
 
-pub fn get_default_file_name(extension: EncodingFormat) -> PathBuf {
-    let current_datetime: DateTime<Local> = Local::now();
-    let formated_time = format!("{}", current_datetime.format("%Y_%m_%d-%H_%M_%S"));
+pub fn get_checked_path(path: &str) -> PathBuf {
+    let checked_path = shellexpand::full(path);
+
+    if let Ok(checked_path) = checked_path {
+        PathBuf::from(checked_path.into_owned())
+    } else {
+        env::current_dir().unwrap_or_default()
+    }
```

Comment:
Maybe I’m on the losing side here, but I prefer not to error where it's possible not to error no matter what. Warnings are good, but the error _can_ cause a situation, when screenshot data is not created and it's hard to reproduce conditions for that screenshot. This situation could easily been avoided instead of throwing error.
IMO having inconsistent behavior is better, than losing data.

Reply:
Heavily disagree here. I want to know when I'm inputting incorrect data. Like I want to know when a Rust function errors and thus want it to return Result and not the Default. I don't think a warning suffices as sometimes you might even pipe the data and you might not even see stderr because it's running in the background.

Reply:
I'm not sure what would be optimal here, however, I agree with Andreas and there should definitely be some "indication" that we are choosing a separate path

Reply:
> I don't think a warning suffices as sometimes you might even pipe the data and you might not even see stderr because it's running in the background.

This does make sense however which makes me slightly incline towards errors.